src/ucs/datastruct/array.h

```diff
@@ -25,7 +25,8 @@ BEGIN_C_DECLS
     struct { \
         _value_type *buffer; \
         _index_type length; \
-        _index_type capacity; \
+        _index_type capacity : ((CHAR_BIT * sizeof(_index_type)) - 1); \
```

Comment:
can we add some overflow checks that capacity does not grow above _index_type/2 ?

Reply:
sure, good idea indeed!